internal/server/handlers.go

```diff
@@ -22,13 +24,61 @@ import (
 	"github.com/infrahq/infra/internal/server/models"
 	"github.com/infrahq/infra/internal/server/providers"
 	"github.com/infrahq/infra/internal/server/redis"
+	"github.com/infrahq/infra/uid"
 )
 
 type API struct {
 	t          *Telemetry
 	server     *Server
-	migrations []apiMigration
 	openAPIDoc openapi3.T
+	versions   map[routeIdentifier][]routeVersion
+}
+
+type routeVersion struct {
+	version *semver.Version
+	handler func(c *gin.Context)
+}
+
+func addVersionHandler[Req, Res any](a *API, method, path, version string, routeDef route[Req, Res]) {
+	if a.versions == nil {
+		a.versions = make(map[routeIdentifier][]routeVersion)
+	}
+
+	routeDef.routeSettings.omitFromDocs = true
```

Comment:
I'm assuming this is because we would end up with conflicting docs for the route. Ideally we would be generating docs for every version that we release, so I _think_ this makes sense.

Reply:
Ya, we don't want previous versions in the latest docs, but we do want to public previous versions of the docs.